lib/import/brca/providers/bristol/bristol_handler.rb

```diff
@@ -8,66 +6,57 @@ module Bristol
         class BristolHandler < Import::Germline::ProviderHandler
           include Import::Helpers::Brca::Providers::Rvj::RvjConstants
 
-          # rubocop:disable  Metrics/MethodLength
-          # rubocop:disable  Metrics/AbcSize
           def process_fields(record)
             genotype = Import::Brca::Core::GenotypeBrca.new(record)
             genotype.add_passthrough_fields(record.mapped_fields,
                                             record.raw_fields,
                                             PASS_THROUGH_FIELDS)
-            process_cdna_change(genotype, record)
-            add_protein_impact(genotype, record)
             add_organisationcode_testresult(genotype)
-            genotype.add_gene_location(record.mapped_fields['codingdnasequencechange'])
-            add_protein_impact(genotype, record)
-            genotype.add_variant_class(record.mapped_fields['variantpathclass'])
-            genotype.add_received_date(record.raw_fields['received date'])
-            process_genomic_change(genotype, record)
             genotype.add_test_scope(:full_screen)
-            process_test_status(genotype, record)
+            genotype.add_received_date(record.raw_fields['received date'])
             genotype.add_method('ngs')
             res = process_gene(genotype, record)
             res&.each { |cur_genotype| @persister.integrate_and_store(cur_genotype) }
           end
-          # rubocop:enable  Metrics/MethodLength
-          # rubocop:enable  Metrics/AbcSize
 
           def add_organisationcode_testresult(genotype)
             genotype.attribute_map['organisationcode_testresult'] = '698V0'
           end
 
-          def process_negative_record(record, negative_gene, genotypes)
-            duplicated_genotype = Import::Brca::Core::GenotypeBrca.new(record)
+          def process_negative_record(genotype, negative_gene, genotypes)
+            duplicated_genotype = genotype.dup
             duplicated_genotype.add_gene(negative_gene.join)
             duplicated_genotype.add_status(1)
-            duplicated_genotype.add_test_scope(:full_screen)
-            duplicated_genotype.add_passthrough_fields(record.mapped_fields,
-                                                       record.raw_fields,
-                                                       PASS_THROUGH_FIELDS)
             genotypes.append(duplicated_genotype)
           end
 
+          def process_positive_record(genotype, positive_gene, record, genotypes)
+            process_cdna_change(genotype, record)
+            genotype.add_gene_location(record.mapped_fields['codingdnasequencechange'])
```

Comment:
Is this overwriting the add_gene_location in process_cdna_change?

Reply:
Right @lauramccluskey1 , unnecessary code.. removed now in latest commit, thank you !